tests/Order.Service.Tests/UseCases/OrderEventListenerTests.cs

```diff
@@ -0,0 +1,149 @@
+
+        // Start the HostedService manually for the test
+        using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(5));
+        var consumerTask = StartOrderEventConsumerAsync(cts.Token);
+
+        OrderModel? order = null;
+        // Act & Assert - Polling pattern similar to the Java example
+        try
+        {
+            order = await PollForOrderProcessingAsync(expectedOrderId, cts);
+
+            Assert.NotNull(order);
+            // Verify the order properties match expected values
+            Assert.Equal(expectedCustomerId, order.CustomerId);
+            Assert.Equal(OrderStatus.Validated, order.Status);
+            Assert.Equal(expectedProductCount, order.ProductQuantities.Count);
+        }
+        catch (TimeoutException)
+        {
+            Assert.Fail("The expected Order was not received/processed in expected delay");
+        }
+        finally
+        {
+            // Stop the consumer - if not already stopped
+            cts.Cancel();
+            try
+            {
+                await consumerTask;
+            }
+            catch (OperationCanceledException)
+            {
+                // Expected when we cancel
+                TestOutputHelper.WriteLine("Order event consumer stopped");
+            }
+        }
+    }
+
+    private async Task<OrderModel> PollForOrderProcessingAsync(
+        string expectedOrderId,
+        CancellationTokenSource cts)
+    {
+        var pollDelay = TimeSpan.FromMilliseconds(400);
+        var pollInterval = TimeSpan.FromMilliseconds(400);
+        var startTime = DateTime.UtcNow;
+
+        var orderUseCase = Factory.Services.GetRequiredService<OrderUseCase>();
+        // Initial delay
+        await Task.Delay(pollDelay, TestContext.Current.CancellationToken);
+
+        while (DateTime.UtcNow - startTime < TimeSpan.FromSeconds(4))
+        {
+            try
+            {
+                // Simulate getting order by ID (this would normally call a GET endpoint)
+                var order = await orderUseCase.GetOrderAsync(expectedOrderId, TestContext.Current.CancellationToken);
+
+                if (order is not null)
+                {
+                    TestOutputHelper.WriteLine($"Order {order.Id} successfully processed!");
+                    cts.Cancel(); // Cancel the consumer after successful processing
+                    return order;
+                }
+            }
+            catch (OrderNotFoundException)
+            {
+                // Continue polling until timeout
+                TestOutputHelper.WriteLine($"Order {expectedOrderId} not found yet, continuing to poll...");
+            }
+
+            await Task.Delay(pollInterval, TestContext.Current.CancellationToken);
+        }
+
+        throw new TimeoutException($"Order {expectedOrderId} was not processed within the expected timeout.");
+    }
+
+    private async Task StartOrderEventConsumerAsync(CancellationToken cancellationToken)
```

Comment:
Why this is to be done explicitly? Couldn't we add this in the `BaseIntegrationTest` setup?